src/common/InputRowRadioButtons.js

```diff
@@ -0,0 +1,154 @@
+import invariant from 'invariant';
+import { CommonActions } from '@react-navigation/native';
+import { BRAND_COLOR, createStyleSheet } from '../styles';
+
+import Touchable from './Touchable';
+import ZulipText from './ZulipText';
+import { IconRight } from './Icons';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+
+type Item<TKey> = $ReadOnly<{|
+  key: TKey,
+  title: string,
+  subtitle?: string,
+|}>;
+
+type Props<TItemKey> = $ReadOnly<{|
+  /**
+   * Component must be under the stack nav with the selectable-options screen.
+   *
+   * Pass this down from props or `useNavigation`.
+   */
+  navigation: AppNavigationProp<>,
+
+  /** What the setting is about, e.g., "Theme". */
+  label: string,
+
+  /** What the setting is about, in a short sentence or two. */
+  description?: string,
+
+  /** The current value of the input. */
+  valueKey: TItemKey,
+
+  /**
+   * The available options to choose from, with unique `key`s, one of which
+   * must be `valueKey`. items: $ReadOnlyArray<Item<TItemKey>>,
+   */
+  items: $ReadOnlyArray<Item<TItemKey>>,
+
+  onValueChange: (newValue: TItemKey) => void,
+|}>;
+
+/**
+ * A form-input row for the user to make a choice, radio-button style.
+ *
+ * Shows the current value (the selected item), represented as the item's
+ * `.title`. When tapped, opens the selectable-options screen, where the
+ * user can change the selection or read more about each selection.
+ */
+// This has the navigate-to-nested-screen semantics of NestedNavRow,
+// represented by IconRight. NestedNavRow would probably be the wrong
+// abstraction, though, because it isn't an imput component; it doesn't have
+// a value to display.
+export default function InputRowRadioButtons<TItemKey: string>(props: Props<TItemKey>): Node {
+  const { navigation, label, description, valueKey, items, onValueChange } = props;
+
+  const screenKey = useRef(
+    `selectable-options-${Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString(36)}`,
+  ).current;
+
+  const selectedItem = items.find(c => c.key === valueKey);
+  invariant(selectedItem != null, 'InputRowRadioButtons: exactly one choice must be selected');
+
+  const screenParams = useMemo(
+    () => ({
+      title: label,
+      description,
+      items: items.map(({ key, title, subtitle }) => ({
+        key,
+        title,
+        subtitle,
+        selected: key === valueKey,
+      })),
+      onRequestSelectionChange: (itemKey, requestedValue) => {
+        if (!requestedValue || itemKey === valueKey) {
+          // Can't unselect a radio button.
+          return;
+        }
+        navigation.goBack();
+        onValueChange(itemKey);
+      },
+    }),
+    [navigation, label, description, items, valueKey, onValueChange],
+  );
+
+  const handleRowPressed = useCallback(() => {
+    // TODO: Can use .push instead of .navigate? See if Flow types are wrong
+    //   in saying we can't pass `key` when using .push.
+    navigation.navigate({
```

Comment:
Hmm, curious.

Even if `push` won't accept a `key`, the behavior is fine, fortunately. I don't recall whether `navigate`'s funky rewind-history behavior (in stack navigators) takes account of `key` to distinguish the desired new destination from existing routes on the stack -- but even if it doesn't, `SelectableOptionsScreen` is not going to push anything else on the stack. That means it'll never be somewhere earlier in the stack when we're here on whatever screen this `InputRowRadioButtons` appears on; so that funky rewinding won't ever trigger here; and the behavior of `navigate` will be exactly the same as that of `push` would have been.

Reply:
Confirmed empirically that `.push` fails when we pass an object with a `key`, in the way that `.navigate` would handle:

<img width=400 src="https://user-images.githubusercontent.com/22248748/167731249-89c5b612-6cd6-4dd2-b273-c52a28974098.png" />

Reply:
So, I'll make a note that using `.navigate` is fine.